src/providers/twitch/eventsub/Controller.cpp

```diff
@@ -154,133 +222,157 @@ SubscriptionHandle Controller::subscribe(const SubscriptionRequest &request)
-                 weakConnection{weakConnection}](const auto &res) {
-                    qCInfo(LOG) << "success" << res;
-                    this->markRequestSubscribed(request, weakConnection,
-                                                res.subscriptionID);
-                },
-                [this, request](const auto &error, const auto &errorString) {
-                    using Error = HelixCreateEventSubSubscriptionError;
-                    switch (error)
-                    {
-                        case Error::BadRequest:
-                            qCWarning(LOG) << "BadRequest" << errorString;
-                            break;
-
-                        case Error::Unauthorized:
-                            qCWarning(LOG) << "Unauthorized" << errorString;
-                            break;
-
-                        case Error::Forbidden:
-                            qCWarning(LOG) << "Forbidden" << errorString;
-                            boost::asio::post(this->ioContext, [this, request] {
-                                this->retrySubscription(
-                                    request, boost::posix_time::seconds(2), 5);
-                            });
-                            break;
-
-                        case Error::Conflict:
-                            // This session ID is already subscribed to this request, some logic of ours is wrong
-                            qCWarning(LOG) << "Conflict" << errorString;
-                            break;
-
-                        case Error::Ratelimited:
-                            qCWarning(LOG) << "Ratelimited" << errorString;
-                            break;
-
-                        case Error::Forwarded:
-                        default:
-                            qCWarning(LOG)
-                                << "Unhandled error, retrying subscription"
-                                << errorString;
-                            boost::asio::post(this->ioContext, [this, request] {
+            qCDebug(LOG) << "New subscribe request for" << request;
+        }
+
+        assert(subscription.retryTimer == nullptr);
+    }
+
+    uint32_t openButNotReadyConnections = 0;
+
+    // 2. Check if any currently open connection can handle this subscription
+    auto viableConnection =
+        this->getViableConnection(openButNotReadyConnections);
+
+    if (viableConnection.has_value())
+    {
+        const auto &connection = *viableConnection;
+        auto *listener = dynamic_cast<Connection *>(connection->getListener());
+
+        qCDebug(LOG) << "Make helix request for" << request;
+        getHelix()->createEventSubSubscription(
+            request, listener->getSessionID(),
+            [this, request, connection,
+             weakConnection{std::weak_ptr<lib::Session>(connection)}](
+                const auto &res) {
+                qCDebug(LOG) << "Subscription success" << request;
+                // TODO: Move marathon stick in here
```

Comment:
Leftover?

Reply:
Fixes in 2dfc02354088bb8e5c3080060fabf770ddf03f7d